### PR TITLE
Fix duplicate dependency issue

### PR DESCRIPTION
**Pull request by davidfowl**
If a package has the same name as a framework assembly in the dependency
graph, we usually replace it with the framework assembly if the package
provides no assets. If the framework assembly wasn't resolved, it would
skip this logic and end up adding dupes to the list, which blows up later on.
This is a tactical fix to solve the issue, we need to do some more thinking
to determine how we want to resolve conflicts between framework assemblies,
packages and dlls with the same name.

#2552

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2597)

### CODE REVIEW

**Comment by troydai** [comment]
Related: https://github.com/dotnet/cli/issues/2602

**Comment by davidfowl** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by pakrym** [comment]
:shipit:

File: test/Microsoft.DotNet.ProjectModel.Tests/PackageDependencyProviderTests.cs [hunk 1]
```diff
@@ -65,5 +65,18 @@ public void TestDuplicateDefaultDesktopReferences(string sampleName, bool withLo
...
+        public void NoDuplicateReferencesWhenFrameworkMissing()
+        {
+            var instance = TestAssetsManager.CreateTestInstance("TestMicrosoftCSharpReferenceMissingFramework")
+                                            .WithLockFiles();
+
+            var context = new ProjectContextBuilder().WithProjectDirectory(instance.TestRoot)
+                                                     .WithTargetFramework("net99")
+                                                     .Build();
+
+            context.LibraryManager.GetLibraries().ToDictionary(l => l.Identity.Name);
```
Comment (eerhardt) [inline test/Microsoft.DotNet.ProjectModel.Tests/PackageDependencyProviderTests.cs]: Should this test assert something?

Comment (pakrym) [inline test/Microsoft.DotNet.ProjectModel.Tests/PackageDependencyProviderTests.cs]: `ToDictionary` is assert that there is no duplications, maybe comment is needed here.

Comment (davidfowl) [inline test/Microsoft.DotNet.ProjectModel.Tests/PackageDependencyProviderTests.cs]: Seems xunit removed `Assert.DoesNotThrow`, which doesn't do anything but it looks nice

**Pull request merged**